packages/ra-core/src/auth/usePermissions.ts

```diff
@@ -13,62 +18,94 @@ const emptyParams = {};
  *
  * The return value updates according to the request state:
  *
- * - start: { isLoading: true }
- * - success: { permissions: [any], isLoading: false }
- * - error: { error: [error from provider], isLoading: false }
+ * - start: { isPending: true }
+ * - success: { permissions: [any], isPending: false }
+ * - error: { error: [error from provider], isPending: false }
  *
  * Useful to enable features based on user permissions
  *
  * @param {Object} params Any params you want to pass to the authProvider
  *
- * @returns The current auth check state. Destructure as { permissions, error, isLoading, refetch }.
+ * @returns The current auth check state. Destructure as { permissions, error, isPending, refetch }.
  *
  * @example
  *     import { usePermissions } from 'react-admin';
  *
  *     const PostDetail = props => {
- *         const { isLoading, permissions } = usePermissions();
- *         if (!isLoading && permissions == 'editor') {
+ *         const { isPending, permissions } = usePermissions();
+ *         if (!isPending && permissions == 'editor') {
  *             return <PostEdit {...props} />
  *         } else {
  *             return <PostShow {...props} />
  *         }
  *     };
  */
-const usePermissions = <Permissions = any, Error = any>(
+const usePermissions = <PermissionsType = any, ErrorType = Error>(
     params = emptyParams,
-    queryParams: UseQueryOptions<Permissions, Error> = {
+    queryParams: UsePermissionsOptions<PermissionsType, ErrorType> = {
         staleTime: 5 * 60 * 1000,
     }
-) => {
+): UsePermissionsResult<PermissionsType, ErrorType> => {
     const authProvider = useAuthProvider();
     const logoutIfAccessDenied = useLogoutIfAccessDenied();
+    const { onSuccess, onError, ...queryOptions } = queryParams ?? {};
 
-    const result = useQuery(
-        ['auth', 'getPermissions', params],
-        authProvider
-            ? () => authProvider.getPermissions(params)
-            : async () => [],
-        {
-            onError: error => {
+    const result = useQuery<PermissionsType, ErrorType>({
+        queryKey: ['auth', 'getPermissions', params],
+        queryFn: () => {
+            return authProvider
+                ? authProvider.getPermissions(params)
+                : Promise.resolve([]);
+        },
+        ...queryOptions,
+    });
+
+    const onSuccessEvent = useEvent(onSuccess ?? noop);
+    const onErrorEvent = useEvent(
+        onError ??
+            (() => {
                 if (process.env.NODE_ENV === 'development') {
-                    console.error(error);
+                    console.error(result.error);
```

Comment:
use the error param instead of result.error

Reply:
Why?